Lacework/lacework_module/lacework_connector.py

```diff
@@ -36,17 +38,32 @@ class LaceworkEventsTrigger(Connector):
 
     def __init__(self, *args: Any, **kwargs: Any) -> None:
         super().__init__(*args, **kwargs)
+        self.from_date = ""
         self.context = PersistentJSON("context.json", self._data_path)
 
     @property
-    def cursor(self) -> str | None:
+    def most_recent_date_seen(self) -> datetime:
+        now = datetime.now(timezone.utc)
+
         with self.context as cache:
-            return str(cache.get("cursor"))
+            most_recent_date_seen_str = cache.get("most_recent_date_seen")
+
+            if most_recent_date_seen_str is None:
+                return now - timedelta(days=1)
+
+            most_recent_date_seen = isoparse(most_recent_date_seen_str)
+
+            one_week_ago = now - timedelta(days=7)
+            if most_recent_date_seen < one_week_ago:
+                most_recent_date_seen = one_week_ago
+
+            return most_recent_date_seen
 
-    @cursor.setter
-    def cursor(self, cursor: str) -> None:
+    @most_recent_date_seen.setter
+    def most_recent_date_seen(self, recent_date: str) -> None:
+        self.from_date = recent_date
```

Comment:
`from_date` must be the greatest date seen. Add a comparison to only save increasing dates
```suggestion
        # save the greatest date
        if self.from_date < recent_date:
                self.from_date = recent_date
```